libsql-sqlite3/src/wal.c

```diff
@@ -4129,13 +4129,12 @@ static int sqlite3WalCheckpoint(
     */
     if( eMode!=SQLITE_CHECKPOINT_PASSIVE ){
       rc = walBusyLock(pWal, xBusy2, pBusyArg, WAL_WRITE_LOCK, 1);
-      if( rc==SQLITE_OK ){
-        pWal->writeLock = 1;
-      }else if( rc==SQLITE_BUSY ){
-        eMode2 = SQLITE_CHECKPOINT_PASSIVE;
-        xBusy2 = 0;
-        rc = SQLITE_OK;
+      if( rc!=SQLITE_OK ){
+        walUnlockExclusive(pWal, WAL_CKPT_LOCK, 1);
+        pWal->ckptLock = 0;
+        return rc;
```

Comment:
If on line 4118 already `rc != SQLITE_OK` then the code continues to do the following things:

```
  walDisableBlocking(pWal);
  sqlite3WalDb(pWal, 0);

  /* Release the locks. */
  sqlite3WalEndWriteTransaction(pWal);
  if( pWal->ckptLock ){
    walUnlockExclusive(pWal, WAL_CKPT_LOCK, 1);
    pWal->ckptLock = 0;
  }
```

Are you sure we don't just want to omit the `else if` and let the code fall through?

Reply:
I changed to fallback already :)

Reply:
ok i am slowpoke